File: solutions/0217-contains-duplicate/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const st = new Set();
    for (const num of nums) {
        if (st.has(num)) return true;
        st.add(num);
    }
    return false;
};

File: solutions/0217-contains-duplicate/solution.java
/*
1. use hashset for this
2. iterate over nums
3. if our hashset contains current num we return true
4. otherwise, add current num to hashset
5. after we iterate we return false by default
*/
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();
        for (int num : nums) {
            if (seen.contains(num)) {
                return true;
            }
            seen.add(num);
        }
        return false;
    }
}
